### ISSUE TITLE
Consider explicitly marking non-injected parameters so they can remain 

### ISSUE THREAD

**Issue by dls314** [nestjs/nest#847]
## I'm submitting a...
[x ] Feature request

## Current behavior

I can't write a class constructor of a class that will be injected where only some of the parameters are understood by nest's injection system.

`constructor(@Inject('ConfigService') configService: ConfigService, public readonly s3: S3 = new S3())`

If I do, I get an error like

>Nest can't resolve dependencies of the S3Service (+, ?). Please make sure that the argument at index [1] is available in the current context.

## Expected behavior
`constructor(@Inject('ConfigService') configService: ConfigService,  @NoInject() public readonly s3: S3 = new S3())`

## What is the motivation / use case for changing the behavior?

It's nice to be able to write wrapper class signatures using constructor access modifiers, it would be great if it could co-exist with nest's DI

## Environment

<pre><code>
Nest version: 5.1.0

**Comment by kamilmysliwiec** [comment]
Actually, why don't you create a custom provider?
```
{ 
    provide: 'S3',
    useValue: new S3(),
}
```

**Comment by autaut03** [comment]
@kamilmysliwiec This might be a good solution most of the time, but to me this looks like over-complicating things =/

**Comment by BrunnerLivio** [comment]
@autaut03 can you elaborate on why this makes it overcomplicated in your use case? Adding just a constant like this for example:

```JavaScript
const S3Provider = { 
    provide: 'S3',
    useValue: new S3(),
}
```

... for each class does not overcomplicate things, does it?

**Comment by autaut03** [comment]
@BrunnerLivio Disclaimer: I'm new to Nest and TS, so please correct me if I'm wrong: 

Well, ```constructor(public readonly s3: S3 = new S3())``` should be the same as 
```readonly s3: S3;

constructor() {
    this.s3 = new S3();
}```

In your example it's 

```readonly s3: S3;

constructor(s3: S3) {
    this.s3 = s3;
}```

where Module is responsible for creating an instance of S3. To me this looks kind of wierd.

Also, again, correct me if I'm wrong, but Nest's deps are all singletons, meaning your example's behaviour is not the same as author's.

**Comment by BrunnerLivio** [comment]
Yes I think factories is the way to go. You loose the power of the injection pattern if you allow to not-inject dependencies. I think something like `@NoInject` should not be allowed, because people tend to use a anti-pattern in the end.

Angular has proven most problem can be solved with this injection pattern, otherwise they would have probably changed their architecture.

**Comment by tmtron** [comment]
here's a link to the [@Optional docs](https://docs.nestjs.com/providers#optional-providers)
